### ISSUE TITLE
Some files remain in released state after "connection reset by peer" error

### ISSUE THREAD

**Issue by simonlindblad** [wastore/lemur#8]
When restoring a dataset (1493347MB/269 blobs) by calling `lfs hsm_restore <file>` for all files, the operation often fails for a subset of them. It doesn't happen every time, but when it does the files that remain in released state (i.e. failed to be restored) have this corresponding error message in the logs:

`
2020/12/04 19:47:37 lhsm-plugin-az2-stderr ALERT 2020/12/04 19:47:37 /usr/src/lemur_wastore/dmplugin/dmclient.go:183: fail: id:4 az.Download() of <filename> failed: read tcp 10.6.6.8:55664->20.150.51.68:443: read: connection reset by peer`

Would it be possible to add retries etc to make it more robust against handling failures from blob storage?

I'm able to reproduce it with different datasets, as long as there are a few files with with a size of several GB.

**Comment by silindbl-msft** [comment]
The config used:
`num_threads=128
az_storage_account="<storage_account>"
az_storage_key="<storage_key>"
region="eastus2"
bandwidth=0
exportprefix=""
archive "archive1" {
    id=1
    num_threads=128
    root=""
    compression="off"
    container="<container_name>"
    region="eastus2"
}`

**Comment by simonlindblad** [comment]
The config used:

```
az_storage_account="<storage_account>"
az_storage_key="<storage_key>"
region="eastus2"
bandwidth=0
exportprefix=""
archive "archive1" {
    id=1
    num_threads=128
    root=""
    compression="off"
    container="<container_name>"
    region="eastus2"
}
```

**Comment by zezha-msft** [comment]
@nakulkar-msft I vaguely remember a similar issue from AzCopy, where certain network errors weren't retried, and we solved it by writing a dedicated retry policy inside AzCopy. Perhaps this is another manifestation of the same problem.

**Comment by simonlindblad** [comment]
Closing this issue, fixed in c8bc6ac6dd06d68deecfa7667b6eb0f842348d44
